File: schemas/helpdesk-customer-support/drizzle/ticket_categories.ts
// ticket_categories: hierarchical topic categories for structured ticket classification.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, boolean, timestamp, index } from "drizzle-orm/pg-core";

export const ticketCategories = pgTable(
  "ticket_categories",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    name: text("name").notNull(),
    slug: text("slug").notNull().unique(),
    description: text("description"),
    parentId: uuid("parent_id").references(() => ticketCategories.id, { onDelete: "set null" }),
    sortOrder: integer("sort_order").notNull().default(0),
    isActive: boolean("is_active").notNull().default(true),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
    updatedAt: timestamp("updated_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_ticket_categories_parent_id_sort_order").on(table.parentId, table.sortOrder),
  ]
);

File: schemas/helpdesk-customer-support/spacetimedb/ticket_categories.rs
// ticket_categories: Hierarchical classification for tickets.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = ticket_categories, public)]
pub struct TicketCategory {
    #[primary_key]
    pub id: String, // UUID
    pub name: String,
    #[unique]
    pub slug: String,
    pub description: Option<String>,
    #[index(btree)]
    pub parent_id: Option<String>, // UUID — FK → ticket_categories.id (set null)
    pub sort_order: i32,
    // Composite index: (parent_id, sort_order) — not supported inline, enforce in application
    pub is_active: bool,
    pub created_at: Timestamp,
    pub updated_at: Timestamp,
}
